### PR TITLE
RestProxy: Removing code that reflect on OperationStatus type in unreleased management-core

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java [hunk 1]
```diff
@@ -375,38 +374,13 @@ private static Object convertToResultType(Object wireResponse, Type resultType,
      */
     private static Type extractEntityTypeFromReturnType(HttpResponseDecodeData decodeData) {
         Type token = decodeData.getReturnType();
-        final ClientLogger logger = new ClientLogger(HttpResponseBodyDecoder.class);
         if (token != null) {
             if (TypeUtil.isTypeOrSubTypeOf(token, Mono.class)) {
                 token = TypeUtil.getTypeArgument(token);
-            } else if (TypeUtil.isTypeOrSubTypeOf(token, Flux.class)) {
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java]: Do we need to handle Flux still?

Comment (anuchandy) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java]: Not in this path. We handle "Flux<ByteBuffer>" in different code path.

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java]: Ah, perfect!

**Pull request merged**